tests/checkpoint/test_checkpoint.py

```diff
@@ -1002,3 +1002,50 @@ def test_is_added(
 
     assert checkpoint_added is is_added
     assert [type(err) for err in errors] == error_list
+
+
+@pytest.mark.unit
```

Comment:
Eh is this really a unit test? It's super fast sure but it also didn't feel like e2e

Reply:
I think it's fine to mark it as a unit test 😄